web/packages/teleport/src/Discover/Shared/ConnectionDiagnostic/useConnectionDiagnostic.ts

```diff
@@ -20,55 +20,89 @@ import useAttempt from 'shared/hooks/useAttemptNext';
 import useTeleport from 'teleport/useTeleport';
 import { useDiscover } from 'teleport/Discover/useDiscover';
 import { DiscoverEventStatus } from 'teleport/services/userEvent';
+import { getDatabaseProtocol } from 'teleport/Discover/Database/resources';
 
 import type {
   ConnectionDiagnostic,
   ConnectionDiagnosticRequest,
 } from 'teleport/services/agents';
-import type { AgentStepProps } from '../../types';
+import type { MfaAuthnResponse } from 'teleport/services/mfa';
+import type { Database } from 'teleport/Discover/Database/resources';
 
-export function useConnectionDiagnostic(props: AgentStepProps) {
+export function useConnectionDiagnostic() {
   const ctx = useTeleport();
 
-  const { attempt, run } = useAttempt('');
+  const { attempt, setAttempt, handleError } = useAttempt('');
   const [diagnosis, setDiagnosis] = useState<ConnectionDiagnostic>();
   const [ranDiagnosis, setRanDiagnosis] = useState(false);
-  const { emitErrorEvent, emitEvent } = useDiscover();
+  const { emitErrorEvent, emitEvent, prevStep, nextStep, resourceState } =
+    useDiscover();
 
   const access = ctx.storeUser.getConnectionDiagnosticAccess();
   const canTestConnection = access.create && access.edit && access.read;
 
-  function runConnectionDiagnostic(req: ConnectionDiagnosticRequest) {
+  const [showMfaDialog, setShowMfaDialog] = useState(false);
+
+  // runConnectionDiagnostic will initially make a call to check if
+  // resource target requires MFA authentication. After this initial
+  // check depending on if user successfully authenticated or not (
+  // determined by the presence of the token field), will make a call
+  // to test connection.
+  //
+  // Each test connection request will require a MFA check since the
+  // fields for the request may have changed eg: for nodes, the login
+  // field may change.
```

Comment:
I deleted this comment. i put it there because i made the initial mistake thinking that one request to `IsMFARequired` was all we needed, but in fact, each request for test connection required its own check because a user can change the request fields to test out different combos for the same resource: https://github.com/gravitational/teleport/blob/b798144dbb1ceaea1e29c256f294bbd8ddc893eb/lib/auth/auth.go#L4082